packages/core/src/Modifiers.ts

```diff
@@ -18,123 +26,162 @@ export class Modifiers extends Array<InlineFormat> {
     //--------------------------------------------------------------------------
 
     /**
-     * Append one or more formats to the array. If one of the given formats is a
-     * format class constructor, instantiate it.
+     * Append one or more modifiers to the array. If one of the given modifiers
+     * is a modifier class constructor, instantiate it.
      *
-     * @param formats
+     * @param modifiers
      */
-    append(...formats: Array<InlineFormat | Constructor<InlineFormat>>): void {
-        for (const format of formats) {
-            if (format instanceof InlineFormat) {
-                this.push(format);
+    append(...modifiers: Array<Modifier | Constructor<Modifier>>): void {
+        for (const modifier of modifiers) {
+            if (modifier instanceof Modifier) {
+                this.push(modifier);
             } else {
-                this.push(new format());
+                this.push(new modifier());
             }
         }
     }
     /**
-     * Prepend one or more formats to the array. If one of the given formats is
-     * a format class constructor, instantiate it.
+     * Prepend one or more modifiers to the array. If one of the given modifiers
+     * is a modifier class constructor, instantiate it.
      *
-     * @param formats
+     * @param modifiers
      */
-    prepend(...formats: Array<InlineFormat | Constructor<InlineFormat>>): void {
-        for (const format of formats) {
-            if (format instanceof InlineFormat) {
-                this.unshift(format);
+    prepend(...modifiers: Array<Modifier | Constructor<Modifier>>): void {
+        for (const modifier of modifiers) {
+            if (modifier instanceof Modifier) {
+                this.unshift(modifier);
             } else {
-                this.unshift(new format());
+                this.unshift(new modifier());
             }
         }
     }
     /**
-     * Return the first format in the array that is an instance of the given
-     * format class, if any. If the format passed is a format instance, return
-     * it if it was present in the array.
+     * Return the first modifier in the array that is an instance of the given
+     * modifier class, if any. If the modifier passed is a modifier instance,
+     * return it if it was present in the array.
      *
-     * @param format
+     * @param modifier
      */
-    get<T extends InlineFormat>(format: T | Constructor<T>): T {
-        if (format instanceof InlineFormat) {
-            return this.find(instance => instance === format) as T;
+    get<T extends Modifier>(modifier: T | Constructor<T>): T {
+        if (modifier instanceof Modifier) {
+            return this.find(instance => instance === modifier) as T;
         } else {
+            const exactMatch = this.find(
+                instance => instance.constructor.name === modifier.name,
+            ) as T;
+            if (exactMatch) return exactMatch;
```

Comment:
Why do we need this particular behavior ? What alternative do we have ?

Reply:
Say I have a ListItemAttributes and a Attributes. I want to apply
Attributes to the list item's contents, and ListItemAttributes to the list
item. Without these lines, if I do .get(Attributes), I'll randomly get
whichever one comes first, of ListItemAttributes and Attributes, because
they both descend from the Attributes class. I need to know that when I do
.get(Attributes) I'll get Attributes in priority.
Granted it's not fantastic but so fat that's all I got.

An alternative is to do `getAll` and then filter the results to find the one with constructor.name ===

Reply:
Ok I got it. Now I don't understand the first `if` then. It will return the instance you gave it so why even bother ? Calling `includes` would work in this case, right ? So, getting rid of the first `if` it leaves us with this:
```typescript
get<T extends Modifier>(Modifier: Constructor<T>): T {
    return (
        this.find(mod => mod.constructor.name === Modifier.name) ||
       this.find(mod => mod instanceof Modifier)
    );
}
```